src/contexts/locales.jsx

```diff
@@ -0,0 +1,77 @@
+/* eslint-env browser */
+import React, { createContext, useEffect } from 'react'
+import useLocalStorage from 'react-use/lib/useLocalStorage'
+import PropTypes from 'prop-types'
+
+const getFromPath = (data, path) => path.split('.').reduce(
+  (curr, sub) => curr && curr[sub],
+  data,
+)
+
+const Context = createContext()
+
+const LocalesProvider = ({ children }) => {
+  const [lang, setLang] = useLocalStorage('lang', undefined)
+  const [locales, setLocales] = useLocalStorage('locales', undefined)
+
+  useEffect(() => {
+    // load lang
+    let currentLang = lang
+    if (lang === undefined) {
+      // retrieve lang from navigator
+      const { navigator } = window
+      if (navigator) {
+        const { language, userLanguage, languages } = navigator
+        currentLang = language || userLanguage || (languages && languages.length > 0 && languages[0]) || 'en'
+        if (currentLang.length >= 2) currentLang = currentLang.substring(0, 2)
+        else currentLang = 'en'
```

Comment:
is this else statement really relevant ?
why would we have a single char retrieved from the navigator infos ?

Reply:
I don't know I prefer to have something robust here because if we don't have locales, nothing will be printed.